CTRP_joined.Rmd

```diff
@@ -0,0 +1,108 @@
+---
+title: "CTRP_joined"
+author: "Yujia Liu"
+date: "8/10/2020"
+output: github_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+# Joining CTRP dataset
+
+## Introduction by Robert
+
+CTRP is a dataset I use in my research. CTRP screens cancer cells with various compounds and determines the Area Under the Dose Response Curve (AUC, a metric for how sensitive a cell line is to that compound). However, when you download the data, it comes in parts that requires a lot of id matching to be able to use reasonably easily. For example, the file that contains the AUC data doesn't tell you the cell line name or the type of cancer it is, it just gives an experiment ID number. If you wanted to know which cell line was which, you'd have to reference the experiment_info.csv file which gives you a cell line ID for each experiment, and then reference the the cancer_cell_line_file.csv to figure out what the name of that cell line actually is. 
+
+```{r library, warning = FALSE, message = FALSE}
+library(tidyverse)
+library(magrittr)
+library(knitr)
+```
+
+## Join separated tables
+
+First, load all 5 tables:
+
+```{r load}
+auc_1 <- read.csv("CTRP_files/AUC_1.csv")
+kable(head(auc_1))
+
+auc_2 <- read.csv("CTRP_files/AUC_2.csv")
+kable(head(auc_2))
+
+cancer_cell_line <- read.csv("CTRP_files/cancer_cell_line_info.csv")
+kable(head(cancer_cell_line))
+
+compound <- read.csv("CTRP_files/compound_info.csv")
+kable(head(compound))
+
+experiment <- read.csv("CTRP_files/Experiment_info.csv")
+kable(head(experiment))
+```
+
+## Join the tables
+
+```{r join}
+ctrp_joined <- experiment %>%
+  # I didn't know more than 2 tables can be joined!
+  left_join(auc_1, auc_2, by = c("expt_id" = "experiment_id")) %>%
+  left_join(cancer_cell_line, by = "master_ccl_id") %>%
+  left_join(compound, by = "master_cpd_id") %>%
+  glimpse()
+```
+
+## Answer the following questions
+
+### Which cancer type has the lowest AUC values to the compound "vorinostat"?
+
+```{r Q1, warning = FALSE}
+ctrp_joined %>%
+  filter(cpd_name == "vorinostat") %>%
+  group_by(cancer_type) %>%
+  # We use the median, in the case of multiple observations per cancer
+  summarize(med_auc = median(area_under_curve)) %>%
+  top_n(10, desc(med_auc)) %>%    # keep top 10
+  ggplot(aes(x = reorder(cancer_type, med_auc), y = med_auc)) +
+    geom_col() +
+    labs(x = "cancer cell line names", y = "median AUC", 
+         title = "Cancer types with the lowest AUC to vorinostat") +
+    theme(axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1))
+```
+
```

Comment:
I think median is a good representation of the data in this case, and your code works well to provide that information. To be honest in my code, I mainly used a boxplot because it is a format I am more familiar with. It did show more information, but I am not sure how useful that was in the long run and it does require more interpretation. I think your methods answer the question appropriately